analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ParameterTypeShouldMatchRouteTypeConstraint.razor

```diff
@@ -0,0 +1,73 @@
+@page "/route/{BoolParam:bool}"
+@page "/route/{BoolParam:int}"
+@page "/route/{BoolParam:int}/{DatetimeParam:datetime}"
+@page "/route/{DatetimeParam:datetime}"
+@page "/route/{DecimalParam:decimal}"
+@page "/route/{DoubleParam:double}"
+@page "/route/{FloatParam:float}"
+@page "/route/{GuidParam:guid}"
+@page "/route/{IntParam:int}"
+@page "/route/{LongParam:long}"
+@page "/route/{StringParam:string}"
+@page "/route/{OptionalStringParam:string?}"
+@page "/route/{CompliantBoolParam:bool}"
+@page "/route/{CompliantDatetimeParam:datetime}"
+@page "/route/{CompliantDecimalParam:decimal}"
+@page "/route/{CompliantDoubleParam:double}"
+@page "/route/{CompliantFloatParam:float}"
+@page "/route/{CompliantGuidParam:guid}"
+@page "/route/{CompliantIntParam:int}"
+@page "/route/{CompliantLongParam:long}"
+@page "/route/{CompliantStringParam:string}"
+@page "/route/{OptionalCompliantStringParam:string?}"
+@page "/route/{InvalidConstraint:invalid}"
```

Comment:
Those are not supported in Blazor (thankfully in a way).
But it might be the case at some point.

For info, the application (Blazor server) does not even start:
> ArgumentException: Unsupported constraint 'range(0, 100)' in route '/counter/{InitialCount:int:range(0, 100)}'.